liquibase-core/src/main/java/liquibase/database/core/PostgresDatabase.java

```diff
@@ -112,52 +109,95 @@ public String getDatabaseChangeLogLockTableName() {
+        if (searchPaths == null/*in case of failure*/ || searchPaths.contains(databaseConnection.getSchema())) {
+            return;
+        }
+        searchPaths.add(databaseConnection.getSchema());
+        String dbName;
+        try {
+            dbName = databaseConnection.getCatalog();
+        } catch (DatabaseException databaseException) {
+            log.warning(("Couldn't get name from DatabaseConnection due to exception"), databaseException);
+            return;
+        }
+        final String ALTER_SEARCH_PATH_QUERY = buildAlterSearchPathQuery(dbName, searchPaths);
+        try (Statement statement = ((JdbcConnection) databaseConnection).createStatement())  {
+            statement.executeUpdate(ALTER_SEARCH_PATH_QUERY);
+            log.info("Default schema: %s has been added to search paths successfully!");
+        } catch (SQLException | DatabaseException exception) {
+            log.warning(String.format("Schema:%s wasn't added to search path due to " +
+                    "exception during execution of SQL statement:%s", databaseConnection.getSchema(), ALTER_SEARCH_PATH_QUERY), exception);
+        }
+    }
+
+    private List<String> getAllSearchPaths(JdbcConnection databaseConnection) {
+        final String SHOW_SEARCH_PATH_QUERY = "SHOW search_path";
+        final List<String> searchPaths = new ArrayList<>();
+        try (Statement statement = databaseConnection.createStatement();
+             ResultSet  resultSet = statement.executeQuery(SHOW_SEARCH_PATH_QUERY)) {
+            if (resultSet.next()) {
+                String searchPath = resultSet.getString(1);
+                List<String> schemas = Arrays.asList(searchPath.split(","));
+                for (String schema: schemas) {
+                    searchPaths.add(schema.trim());
+                }
+            }
+        } catch (SQLException | DatabaseException exception) {
+            log.warning(String.format("Statement:%s couldn't be executed due to exception", SHOW_SEARCH_PATH_QUERY), exception);
+            return null;
+        }
+        return searchPaths;
+    }
 
-        Logger log = LogFactory.getInstance().getLog();
+    private String buildAlterSearchPathQuery(String dbName, List<String> searchPaths) {
+        return new StringBuilder().append(" ALTER DATABASE ").
+                     append(dbName).
+                     append(" SET ").
+                     append(" search_path TO ").
+                     append(StringUtils.join(searchPaths, ",")).toString();
+    }
+
+    /**
+     * In this method we log.warn If edb_redwood_date is set to TRUE. Specific to PostgreSQL
+     * @param databaseConnection - jdbc connection that will use in order to execute statement
+     */
+    private void logWarnIfStoreDateColumnNotSupported(final DatabaseConnection databaseConnection) {
+        if (databaseConnection instanceof JdbcConnection) {
+            try (Statement statement =((JdbcConnection) databaseConnection).createStatement();
```

Comment:
We can't use the try-with syntax in java 6

Reply:
I didn't know. I'll refactor it

Reply:
Refactored. I've a question.
Why liquebase still on Java 6 ?